Sources/SyrupCore/GraphQL/IntermediateRepresentationVisitor.swift

```diff
@@ -86,7 +88,15 @@ class IntermediateRepresentationVisitor: GraphQLBaseVisitor {
 		let name = operation.name!
 		let selections = parsedSelections.pop()
 		parentType.pop()
-		let type: IntermediateRepresentation.OperationType = operation.operationType == .query ? .query : .mutation
+		var type: IntermediateRepresentation.OperationType
```

Comment:
Not sure if there's a cleaner way to do this in Swift

Reply:
It can be left as a `let` and not a `var` if the switch below will always set the value before it's used

Reply:
Maybe the assignment could be in the switch statement below and switched on `operation.operationType`?

```swift
let type: IntermediateRepresentation.OperationType
let operation: String
let typeName: String
switch operation.operationType {
case .query:
	type = .query
	operation = queries[name]!
	typeName = schema.queryType.name
 case .mutation:
	type = .mutation
 	operation = mutations[name]!
 	typeName = schema.mutationType!.name
 case .subscription:
	type = .subscription
 	operation = subscriptions[name]!
 	typeName = schema.subscriptionType!.name
 }
```